provider/pkg/provider/provider_2e2_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
+
+package provider_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/pulumi/providertest"
+	"github.com/pulumi/providertest/pulumitest"
+	"github.com/pulumi/providertest/pulumitest/assertpreview"
+	"github.com/pulumi/providertest/pulumitest/assertrefresh"
+	"github.com/pulumi/providertest/pulumitest/opttest"
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/provider"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+)
+
+func TestE2eSnapshots(t *testing.T) {
+	t.Run("logGroup-0.94.0", func(t *testing.T) {
+		t.Parallel()
+		test := newAwsTest(t, filepath.Join("testdata", "logGroup"))
+		testUpgradeFrom(test, "0.94.0")
+	})
+
+	t.Run("webAcl-0.94.0", func(t *testing.T) {
+		t.Parallel()
+		test := newAwsTest(t, filepath.Join("testdata", "webAcl"))
+		testUpgradeFrom(test, "0.94.0")
+	})
+}
+
+func testUpgradeFrom(test *pulumitest.PulumiTest, version string) {
+	test.T().Helper()
+	result := providertest.PreviewProviderUpgrade(test, "aws-native", version)
```

Comment:
does this helper include all the logic to record state too?

Reply:
Yup - there's options to control the state storage but the defaults will store it in the same place as the old version of upgrade test.

Reply:
Oh, here's the docstring for reference:
> PreviewProviderUpgrade captures the state of a stack from a baseline provider configuration, then previews the stack with the current provider configuration. Uses a default cache directory of "testdata/recorded/TestProviderUpgrade/{programName}/{baselineVersion}".